cfme/tests/infrastructure/test_snapshot.py

```diff
@@ -94,21 +98,28 @@ def test_delete_all_snapshots(test_vm, provider):
     snapshot2.delete_all()
 
 
-@pytest.mark.meta(blockers=[1333566])
-@pytest.mark.uncollectif(lambda provider: provider.type != 'virtualcenter')
+@pytest.mark.uncollectif(lambda provider:
+    not provider.one_of(RHEVMProvider, VMwareProvider) or
+    (provider.one_of(RHEVMProvider) and provider.version < 4))
 def test_verify_revert_snapshot(test_vm, provider, soft_assert, register_event, request):
     """Tests revert snapshot
 
     Metadata:
         test_flag: snapshot, provision
     """
-    snapshot1 = new_snapshot(test_vm)
+    if provider.one_of(RHEVMProvider):
+        snapshot1 = new_snapshot(test_vm, has_name=False)
+    else:
+        snapshot1 = new_snapshot(test_vm)
     ip = snapshot1.vm.provider.mgmt.get_ip_address(snapshot1.vm.name)
-    ssh_kwargs = {
-        'username': credentials[provider.data['full_template']['creds']]['username'],
-        'password': credentials[provider.data['full_template']['creds']]['password'],
-        'hostname': ip
-    }
+    ssh_kwargs = {'hostname': ip}
+    if provider.one_of(RHEVMProvider):
+        ssh_kwargs['username'] = credentials[provider.data['ssh_creds']]['username']
+        ssh_kwargs['password'] = credentials[provider.data['ssh_creds']]['password']
+    else:
+        ssh_kwargs['username'] = credentials[provider.data['full_template']['creds']]['username']
+        ssh_kwargs['password'] = credentials[provider.data['full_template']['creds']]['password']
```

Comment:
I think there is a bit of a confusion here about ssh credentials.
`provider.data['ssh_creds']` = credentials key to access the provider by SSH
`provider.data['full_template']['creds']` = credentials key to access the template, when provisioned into a VM